middleware/authenticate.js

```diff
@@ -1,32 +1,71 @@
-// const jwt = require('jsonwebtoken');
-// const {User} = require("../services/authService");
-// const express = require('express')
-// const {header} = require("express-validator");
-// const authJwt = new express.Router();
-//
-// // authJwt.post('/refreshToken', async (req, res) => {
-// //     const refreshToken = req.body.refreshToken;
-// //     if (refreshToken == null) return res.sendStatus(401)
-// //     // const user = await User.retrieve({refreshToken: refreshToken})
-// //     // if (!user) return res.sendStatus(403)
-// //     jwt.verify(refreshToken, process.env.REFRESH_TOKEN_SECRET, (err, user) => {
-// //         if (err) return res.sendStatus(403)
-// //         const accessToken = generateAccessToken({email: user.email})
-// //         //  res.setHeader("authorization", "Bearer" + " " + accessToken)
-// //         req.token = {accessToken: accessToken};
-// //         res.redirect(req.originalUrl);
-// //     })
-// // })
-//
-// authJwt.delete('/logout', (req, res) => {
-//     const refreshTokens = req.body.token
-//     User.delete({refreshTokens: refreshTokens})
-//     res.sendStatus(204)
-// })
-//
-// function generateAccessToken(user) {
-//     return jwt.sign(user, process.env.ACCESS_TOKEN_SECRET, {expiresIn: '15m'})
-// }
-//
-//
-// module.exports = {authJwt};
+const jwt = require('jsonwebtoken');
+const { User } = require('../services/authService');
+
+const generateToken = (req, res, next) => {
+  try {
+    const accessToken = jwt.sign({
+      email: req.body.email,
+    }, process.env.ACCESS_TOKEN_SECRET, {
+      expiresIn: '5s',
+    });
+
+    const refreshToken = jwt.sign({
+      email: req.body.email,
+    }, process.env.REFRESH_TOKEN_SECRET, { expiresIn: '1d' });
+    res.cookie('jwt', refreshToken, {
+      httpOnly: true,
+      sameSite: 'None',
+      secure: true,
+      maxAge: 24 * 60 * 60 * 1000,
+    });
+
+    res.set({ authorization: `Bearer ${accessToken}` });
+    req.token = { refreshToken, accessToken: `Bearer ${accessToken}` };
+    next();
+  } catch (e) {
+    return res.sendStatus(401);
+  }
+};
+
+function generateAccessToken(user) {
+  return jwt.sign(user, process.env.ACCESS_TOKEN_SECRET, { expiresIn: '5s' });
+}
+
+const refreshTokenVerify = async (req, res) => {
+  const refreshToken = req.cookies.jwt;
+  if (refreshToken == null) return res.status(403).json({ message: 'Unauthorized' });
+
+  const user = await User.retrieve({ refreshToken });
```

Comment:
`register.otpCompare(req.body.code, oneTimePassRecord.code);`
what if this throws?